src/main/java/com/thealgorithms/sorts/PigeonholeSort.java

```diff
@@ -1,55 +1,88 @@
 package com.thealgorithms.sorts;
 
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
-public class PigeonholeSort {
+public final class PigeonholeSort {
+    private PigeonholeSort() {
+    }
 
-    /*
-        This code implements the pigeonhole sort algorithm for the integer array,
-        but we can also implement this for string arrays too.
-        See https://www.geeksforgeeks.org/pigeonhole-sort/
-    */
-    void sort(Integer[] array) {
-        int maxElement = array[0];
-        for (int element : array) {
-            if (element > maxElement) {
-                maxElement = element;
-            }
+    /**
+     * Sorts the given array using the pigeonhole sort algorithm.
+     *
+     * @param array the array to be sorted
+     * @throws IllegalArgumentException if any negative integers are found
+     * @return the sorted array
+     */
+    public static int[] sort(int[] array) {
+        if (array.length < 2) {
+            return array;
```

Comment:
```suggestion
        checkForNegativeInput(array);

        if (array.length == 0) {
            return array;
```

Two things:
- the check for negative numbers has to happen first,
- it seems enough to check if the array is empty.

Reply:
Thanks.  Thanks. Input checking for empty array will be more clear than checking for exact size.

Is this way to check negative input first is better from code clarity and readability perspective? To highlight first that it doesn't work with negative numbers?